stumpy/gpu_stump.py

```diff
@@ -7,17 +7,110 @@
 import os
 
 import numpy as np
-from numba import cuda
+from numba import cuda, jit
```

Comment:
> Have you considered what it means `k > 1` when dealing with an AB-join? I'm not entirely sure what the meaning is OR if it should even be allowed?

Yes. I did. So, this is just my thought. Please correct me if I am wrong. I think the main difference between `self-join` and `AB-join` is that we ignore trivial matches in the former. So, the concept of `Top-k NearestNeighbor(s)` does not change. Therefore, we basically finding top-k nearest neighbors of `seq_of_TA` in `TB` (Please note that, as we discussed before, when we talk about top-k matrix profile here, we do not care whether those neighbors themselves are trivial matches of each other or not. We ignore trivial matches in finding top-k NN. (In other words, we STILL CONSIDER trivial matches in self-join when updating distance profile by filling the trivial matches with `np.inf`). 


ALSO:
In fact, this way of finding top-k NN is required by VALMOD paper. I am not sure but I think VALMOD just discussed self-join. However, the formula provided in the paper (and the derivation we did) show that it can be simply applied to AB-join as well for accelerating matrix profile calculation.

Reply:
I will remove `jit`.